Tasks/JenkinsQueueJob/job.ts

```diff
@@ -98,7 +107,7 @@ export class Job {
             } else if (oldState == JobState.Streaming) {
                 validStateChange = (newState == JobState.Finishing);
             } else if (oldState == JobState.Finishing) {
-                validStateChange = (newState == JobState.Downloading || newState == JobState.Done);
+                validStateChange = (newState == JobState.Downloading || newState == JobState.Queued || newState == JobState.Done);
```

Comment:
The state transition from "Finishing" to "Queued" seems odd?  How could a running job become "Queued" again?